pycontrails/core/met.py

```diff
@@ -1022,6 +1037,84 @@ def to_vector(self, transfer_attrs: bool = True) -> vector_module.GeoVectorDatas
             vector.attrs.update({str(k): v for k, v in self.attrs.items()})
         return vector
 
+    def _get_attr_template(
+        self,
+        name: str,
+        supported: tuple[str, ...],
+        examples: dict[str, str],
+    ) -> str:
+        """Look up an attribute with a custom error message."""
+        try:
+            out = self.attrs[name].upper()
+        except KeyError as e:
+            msg = f"Specify '{name}' attribute on underlying dataset."
+
+            for i, (k, v) in enumerate(examples.items()):
+                if i == 0:
+                    msg = f"{msg} For example, set attrs['{name}'] = '{k}' for {v}."
+                else:
+                    msg = f"{msg} Set attrs['{name}'] = '{k}' for {v}."
+            raise KeyError(msg) from e
+
+        if out not in supported:
+            warnings.warn(
+                f"Unknown {name} '{out}'. Data may not be processed correctly. "
+                f"Known {name}s are {supported}. Contact the pycontrails "
+                "developers if you believe this is an error."
+            )
+
+        return out
+
+    @property
+    def provider_attr(self) -> str:
```

Comment:
What do you think about removing the `attr` suffix from these? 

It works well for `provider` and `product` but we'd probably want to rename `dataset` to `dataset_name` or something?

Reply:
I take it back, I think the `_attr` suffix is probably worth ith

Reply:
Alternatively, if the `attrs` key is changed to `pycontrails_provider`, the property could also be `pycontrails_provider`.